lib/matplotlib/legend_handler.py

```diff
@@ -148,21 +148,24 @@ def get_numpoints(self, legend):
     def get_xdata(self, legend, xdescent, ydescent, width, height, fontsize):
         numpoints = self.get_numpoints(legend)
 
-        if numpoints > 1:
+        if ((numpoints < 1) or not (isinstance(numpoints, int) or numpoints.is_integer())):
```

Comment:
What's the difference between these two?
`isinstance(numpoints, int) or numpoints.is_integer())`